tests/api-controller-acceptance-tests/apicontroller/component_test.go

```diff
@@ -188,6 +188,44 @@ func TestComponentSpec(t *testing.T) {
 			So(policy.Spec, ctx.ShouldDeepEqual, expectedPolicy)
 		})
 
+		Convey("create API should not process the request if another API exists for target service", func() {
+			t.Log("duplicate API for a service")
+
+			testService := "test-srv"
+
+			id := ctx.generateTestID(testIDLength)
+			api := ctx.apiFor(id, domainName, namespace, apiSecurityEnabled, true)
+			api.Spec.Service.Name = testService
+
+			testedID := ctx.generateTestID(testIDLength)
+			testedApi := ctx.apiFor(testedID, domainName, namespace, apiSecurityDisabled, true)
+			testedApi.Spec.Service.Name = testService
+
+			kymaClient.GatewayV1alpha2().Apis(namespace).Create(api)
+			lastAPI, err := kymaClient.GatewayV1alpha2().Apis(namespace).Create(testedApi)
+
+			So(err, ShouldBeNil)
+			So(lastAPI, ShouldNotBeNil)
+			So(lastAPI.ResourceVersion, ShouldNotBeEmpty)
+
+			time.Sleep(5 * time.Second)
+
+			lastAPI, err = kymaClient.GatewayV1alpha2().Apis(namespace).Get(lastAPI.Name, metav1.GetOptions{})
+			So(err, ShouldBeNil)
+			//So(lastAPI.Status.IsTargetServiceOccupied(), ShouldBeTrue)
+			So(lastAPI.Status.AuthenticationStatus.IsEmpty(), ShouldBeTrue)
+			So(lastAPI.Status.VirtualServiceStatus.IsEmpty(), ShouldBeTrue)
+
+			policy, err := istioAuthClient.AuthenticationV1alpha1().Policies(namespace).Get(lastAPI.Status.AuthenticationStatus.Resource.Name, metav1.GetOptions{})
```

Comment:
If policy and virtualService are not created their `Name` in respective status object will be empty. Maybe it is enough to check them instead of querying K8s?